apps/vscode/src/vdoc/vdoc.ts

```diff
@@ -157,12 +157,8 @@ export async function virtualDocUri(
 export function languageAtPosition(tokens: Token[], position: Position) {
   const block = languageBlockAtPosition(tokens, position);
   if (block) {
```

Comment:
IIUC, previously it _attempted_ to update the context key dynamically on each cell "activation", right? That does seem hard to get 100% right.

And now it tries to update the context key 1 time for whatever the main language in the document is.

I suppose that _could_ be mildly confusing. If you have 1 python cell in a mostly R document, then clicking in the python cell and running a keybinding will run the R version. That's kinda confusing, but I'm not sure what we can do better.

Reply:
I did hope for us to get this working really nicely for reticulate Quarto files (per cell), but when I tried to use the approach with `languageAtPosition()` in practice, it felt quite awful. So yep, I think for now this approach with the "main language" is the best option.